include/pybind11/cast.h

```diff
@@ -1998,6 +2053,8 @@ class argument_loader {
         if ((... || !std::get<Is>(argcasters).load(call.args[Is], call.args_convert[Is])))
             return false;
 #else
+        // BUG: When loading the arguments, the actual argument type (pointer, lvalue reference, rvalue reference)
```

Comment:
Yeah, we had run into this as well for our fork :(
We fixed it here: https://github.com/RobotLocomotion/pybind11/pull/11

Basically, I think you can just add a destructor which will allow pybind11 to reclaim ownership?